site/content/docs/contributing/developer-docs/plugin-query-system.md

```diff
@@ -0,0 +1,196 @@
+---
+title: The Hipcheck Plugin Query System
+weight: 2
+---
+
+# The Hipcheck Plugin Query System
+
+This document describes the control flow through Hipcheck core, down to gRPC,
+into the Rust SDK, and back to the core during a plugin system query. This
+document assumes the plugins are already started and configured, and that we
+have established a gRPC stream with them over which to send and receive
+messages defined by our `hipcheck-common/proto` protobuf schema.
+
+#![image info](static/images/developer_docs_plugin_grpc.png)
+
+## Overview and Design Requirements
+
+Hipcheck plugins are child processes of the Hipcheck core process that it
+communicates with over distinct gRPC channels. Each Hipcheck plugin defines a
+set of query endpoints that act as remote functions. They each receive a JSON-
+serialized key and return a JSON-serialized result. During a query endpoint's
+execution, it may need to invoke another plugin's query endpoint(s). All
+communicaton between plugins goes through Hipcheck core, so just as Hipcheck
+core issues a request to a given endpoint, the endpoint can request Hipcheck
+core to issue another request to a different endpoint and report to the original
+endpoint the response, so that the original endpoint can complete its own
```

Comment:
feel free to disregard if this is nitpicky but  "report the response to the original endpoint" sounds a bit more clear